pac4j-core/src/main/java/org/pac4j/core/matching/matcher/csrf/CsrfTokenGeneratorMatcher.java

```diff
@@ -39,12 +40,12 @@ public boolean matches(final WebContext context, final SessionStore sessionStore
         final var token = csrfTokenGenerator.get(context, sessionStore);
         context.setRequestAttribute(Pac4jConstants.CSRF_TOKEN, token);
         final var cookie = new Cookie(Pac4jConstants.CSRF_TOKEN, token);
-        if (domain != null) {
+        if (StringUtils.hasText(domain)) {
```

Comment:
Shiro is an optional dependency, we can't use its `StringUtils` method. Use the pac4j `CommonHelper`.